### DISCUSSION TITLE
Is there a way to prevent multiple instances?

### DISCUSSION THREAD

**Discussion by ozgunozerk** [tauri-apps/tauri#3557] [Q&A] [answered]
Is there a native way to enforce a single instance, or prevent multiple instances for the desktop tauri app?

**Comment by FabianLars** [comment, marked as answer]
Not yet (work on it won't start before v1 stable is released).

Notable community experiments:
https://gitlab.com/ioneyed/tauri-example-singleinstance
https://gitlab.com/ioneyed/tauri-plugin-highlander
Furthermore you can find discussions around this on our discord.

**Comment by ankitpatial** [comment]
Is there any progress on it as tauri is already in v1 from last few months?

**Reply by FabianLars** [reply]
For windows and linux there's this <https://github.com/amrbashir/tauri-plugin-single-instance/>. amr is a team member, so this is somewhat official.
I'm not a macos user, but isn't single-instance the default there? If so this plugin may be enough for most use cases.

**Reply by ankitpatial** [reply]
@FabianLars thank you for quick reply. I guess you are right about mac(I will confirm it)

**Reply by TFLXX** [reply]
> @FabianLars thank you for quick reply. I guess you are right about mac(I will confirm it)

Can you confirm it? 🙂

**Reply by LordNayan** [reply]
@TFLXX I can confirm that mac has single instance by default

**Comment by hugoattal** [comment]
If anyone looking for this, here's the official plugin: https://github.com/tauri-apps/plugins-workspace/tree/v1/plugins/single-instance

**Reply by Karuma303** [reply]
Well, actually I just searched for it this afternoon and also found the official plugin. So this topic seems to be very hot! :)
